readthedocs/core/symlink.py

```diff
@@ -144,26 +144,26 @@ def symlink_cnames(self, domain=None):
         if domain:
             domains = [domain]
         else:
-            domains = Domain.objects.filter(project=self.project)
+            domains = Domain.objects.filter(project=self.project).values_list('domain', flat=True)
         for dom in domains:
-            log_msg = 'Symlinking CNAME: {0} -> {1}'.format(dom.domain, self.project.slug)
+            log_msg = 'Symlinking CNAME: {0} -> {1}'.format(dom, self.project.slug)
             log.info(constants.LOG_TEMPLATE.format(project=self.project.slug,
                                                    version='', msg=log_msg))
 
             # CNAME to doc root
-            symlink = os.path.join(self.CNAME_ROOT, dom.domain)
+            symlink = os.path.join(self.CNAME_ROOT, dom)
             run(['ln', '-nsf', self.project_root, symlink])
 
             # Project symlink
-            project_cname_symlink = os.path.join(self.PROJECT_CNAME_ROOT, dom.domain)
+            project_cname_symlink = os.path.join(self.PROJECT_CNAME_ROOT, dom)
             run(['ln', '-nsf', self.project.doc_path, project_cname_symlink])
 
-    def remove_symlink_cname(self, domain):
+    def remove_symlink_cname(self, domain_str):
```

Comment:
I think the argument should still be called `domain`, as:

* We don't annotate variable names with their types
* The code logic doesn't dictate that it's necessary to change `domain.domain` to anything but `domain`
* Other calls could be calling this as `remove_symlink_cname(domain=...)`